wasmtime/_func.py

```diff
@@ -56,21 +64,39 @@ def type(self, store: Storelike) -> FuncType:
         ptr = ffi.wasmtime_func_type(store._context, byref(self._func))
         return FuncType._from_ptr(ptr, None)
 
-    def __call__(self, store: Storelike, *params: IntoVal) -> Union[IntoVal, Sequence[IntoVal], None]:
-        """
-        Calls this function with the given parameters
-
-        Parameters can either be a `Val` or a native python value which can be
-        converted to a `Val` of the corresponding correct type
-
-        Returns `None` if this func has 0 return types
-        Returns a single value if the func has 1 return type
-        Returns a list if the func has more than 1 return type
+    def _create_raw_vals(self, *params: IntoVal) -> ctypes.Array[wasmtime_val_raw_t]:
+        raw = self._vals_raw_type()
+        for i, param_str in enumerate(self._params_str):
+            val_setter(self._func.store_id, raw[i], param_str, params[i])
+        return raw
 
-        Note that you can also use the `__call__` method and invoke a `Func` as
-        if it were a function directly.
+    def _extract_return(self, vals_raw: ctypes.Array[wasmtime_val_raw_t]) -> Union[IntoVal, Sequence[IntoVal], None]:
+        if self._results_n == 0:
+            return None
+        if self._results_n == 1:
+            return val_getter(self._func.store_id, vals_raw[0], self._results_str0)
+        # we can use tuple construct, but I'm using list for compatability
```

Comment:
This comment can be removed. Either the result should continue to be a list in which case there's no need for the comment, or the result should change to a tuple, in which case there's no need for the comment.